src/lib.rs

```diff
@@ -97,6 +95,14 @@ pub trait GraphDataWriter: Graph {
     fn node_data_mut(&mut self, id: &Id<Self::Node>) -> Option<&mut Data<Self::Node>>;
 }
 
+/// An annotator for graphs.
+pub trait GraphAnnotator {
+    type Annotation;
+
+    /// Annotate the graph with some data.
+    fn annotate_graph(&mut self, note: Self::Annotation);
```

Comment:
Is there a reason why we make the `Annotation` generic? Related, why not use the function `annotate_node(nodeId, value)`?

Reply:
It gives control over the implementation to choose what kind of annotation it should be - since the graph-api traits aren't specific to osrank.

Reply:
Ok, that makes sense. Can we follow this up with a PR for the concrete annotation that the Ledger and Osrank are interested in? That is setting the osrank of projects and accounts.

Reply:
Yep, good idea! I'll create a PR on top of the concrete-types PR.